src/main/java/org/springframework/samples/petclinic/rest/VetRestController.java

```diff
@@ -120,7 +121,11 @@
 		if(vet == null){
 			return new ResponseEntity<Void>(HttpStatus.NOT_FOUND);
 		}
-		this.clinicService.deleteVet(vet);
+        Collection<Visit> allVisits = this.clinicService.findAllVisits();
```

Comment:
As mentioned in the general comment, the business logic must be moved inside the service layer.
Also the delete must be based on cascade type, or orphanDelete = true. For the difference between the two, please check -> https://stackoverflow.com/a/18813411

*Also based on the comment, we must add a collection of visits inside the vet.